src/test/java/org/broadinstitute/hellbender/tools/copynumber/PreprocessIntervalsIntegrationTest.java

```diff
@@ -0,0 +1,146 @@
+        expectedBinsWholeGenomeTest.add(new Interval("20", 30_000_001, 40_000_000));
+        expectedBinsWholeGenomeTest.add(new Interval("20", 40_000_001, 50_000_000));
+        expectedBinsWholeGenomeTest.add(new Interval("20", 50_000_001, 60_000_000));
+        expectedBinsWholeGenomeTest.add(new Interval("20", 60_000_001, 63_025_520));
+        expectedBinsWholeGenomeTest.add(new Interval("21", 1, 10_000_000));
+        expectedBinsWholeGenomeTest.add(new Interval("21", 10_000_001, 20_000_000));
+        expectedBinsWholeGenomeTest.add(new Interval("21", 20_000_001, 30_000_000));
+        expectedBinsWholeGenomeTest.add(new Interval("21", 30_000_001, 40_000_000));
+        expectedBinsWholeGenomeTest.add(new Interval("21", 40_000_001, 48_129_895));
+
+        // Return all test data
+        return new Object[][]{
+                {binLengthSeparateIntervalTest, paddingSeparateIntervalTest, outputFileNameSeparateIntervalTest, inputIntervalsSeparateIntervalTest, expectedBinsSeparateIntervalTest},
+                {binLengthOverlappingIntervalTest, paddingOverlappingIntervalTest, outputFileNameOverlappingIntervalTest, inputIntervalsOverlappingIntervalTest, expectedBinsOverlappingIntervalTest},
+                {binLengthEdgeIntervalTest, paddingEdgeIntervalTest, outputFileNameEdgeIntervalTest, inputIntervalsEdgeIntervalTest, expectedBinsEdgeIntervalTest},
+                {binLengthWholeChromosomeTest, paddingWholeChromosomeTest, outputFileNameWholeChromosomeTest, inputIntervalsWholeChromosomeTest, expectedBinsWholeChromosomeTest},
+                {binLengthWholeGenomeTest, paddingWholeGenomeTest, outputFileNameWholeGenomeTest, inputIntervalsWholeGenomeTest, expectedBinsWholeGenomeTest}
+        };
+    }
+
+    // Test for interval inputs given as -L command line arguments
+    @Test(dataProvider = "intervalInputsFromCommandLine")
+    public final void testCommandLine(final int binLength, final int padding,  final String[] outputFileName, final ArrayList<Interval> inputIntervals, final ArrayList<Interval> expectedBins) {
+        final File outputFile = createTempFile(outputFileName[0], outputFileName[1]);
+        final ArrayList<String> args = new ArrayList<>(Arrays.asList(
+                "-R", REFERENCE.getAbsolutePath(),
+                "-" + PreprocessIntervals.LENGTH_OF_BINS_SHORT_NAME, Integer.toString(binLength),
+                "-" + PreprocessIntervals.PADDING_SHORT_NAME, Integer.toString(padding),
+                "-O", outputFile.getAbsolutePath()
+        ));
+        for (Interval interval : inputIntervals) {
+            args.add("-L");
+            args.add(interval.getContig() + ":" + Integer.toString(interval.getStart()) + "-" + Integer.toString(interval.getEnd()));
+        }
+
+        runCommandLine(args.toArray(new String[0]));
+
+        final IntervalList binsResult = IntervalList.fromFile(outputFile);
+        Assert.assertEquals(binsResult, expectedBins);
```

Comment:
This is a little nitpicky, but I like `binsResult` and `binsExpected` better (and I see that you named them so below).

Reply:
OK